projects/nfdi-gdo-app/src/app/data-selection.service.ts

```diff
@@ -63,8 +63,8 @@ export class DataSelectionService {
         );
     }
 
-    setPolygonLayer(layer: VectorLayer): Observable<void> {
-        return this.polygonLayer.pipe(
+    setSpeciesLayer(layer: VectorLayer): Observable<void> {
+        return this.speciesLayer.pipe(
```

Comment:
For some reason, when I select another species, the view does not update.

Reply:
I removed the clearLayers() call in selectSpecies and it works now